src/main/java/io/github/ImpactDevelopment/installer/setting/settings/OptiFineSetting.java

```diff
@@ -22,53 +22,11 @@
 
 package io.github.ImpactDevelopment.installer.setting.settings;
 
-import io.github.ImpactDevelopment.installer.setting.ChoiceSetting;
-import io.github.ImpactDevelopment.installer.setting.InstallationConfig;
-import io.github.ImpactDevelopment.installer.target.InstallationModeOptions;
+import io.github.ImpactDevelopment.installer.setting.BooleanSetting;
 
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.List;
-import java.util.stream.Collectors;
-import java.util.stream.StreamSupport;
-
-public enum OptiFineSetting implements ChoiceSetting<String> {
+public enum OptiFineSetting implements BooleanSetting {
     INSTANCE;
 
-    public static final String NONE = "None";
-    public static final String MISSING = "Missing";
-
-    @Override
-    public List<String> getPossibleValues(InstallationConfig config) {
-        if (config.getSettingValue(InstallationModeSetting.INSTANCE) == InstallationModeOptions.FORGE || config.getSettingValue(MinecraftVersionSetting.INSTANCE).compareTo("1.15.2") > 0) {
```

Comment:
This might be something good to include. My reasoning was that for 2 of 3 past mc version bumps, optifine changed their shit and the installer broke :(

The thought is it's better to allow installs without optifine than to bork ppl's install with incorrect optifine idk

iiiiiiidk

Reply:
yeah, I hadn't considered the "upgrade an existing install with optifine" scenario... not sure how that'd work with multimc either.

I'm starting to feel like the installer also being an upgrader wasn't a good idea... IMO just telling the user "an update exists" is enough without also installing it for them, other issues I have with it is that the updater isn't particularly configurable, e.g. you cant disable it per se and in some scenarios you can't prevent it from forcing an upgrade either.

Reply:
Honestly, if the part builtin to the client were dumbed down to only have the capability to install patch releases to your current install, I could live with that.